src/unlocked/didcomm.rs

```diff
@@ -22,13 +25,22 @@ impl UnlockedWallet {
     /// * `message` - fully populated JSON serialized `Message` ready for sealing;
     /// * `header` - JSON serialized `DidcommHeader`. Ready for send.
     ///
-    pub fn seal_signed(&self, key_id: &str, sign_key_controller: &str, message: &str, header: &str)
+    pub fn seal_signed(&self, message: &str)
         -> Result<String, Error> {
-            if let Some(kp) = self.contents.get_by_controller(sign_key_controller) {
+            let m: Message = serde_json::from_str(message)?;
+            let document = try_resolve_any(&m.get_didcomm_header().from.clone().unwrap_or_default())
+                .map_err(|e| Error::Other(e.into()))?;
+            let ekp: Option<(String, &Content)> =
+                if let Some(ka) = document.find_key_agreement("Ed25519") {
```

Comment:
Was initially confused by this, thought we are fetching a key agreement key, but after looking into it a bit more I see that this can return different key types (like `JwsVerificationKey2020`, `EcdsaSecp256k1VerificationKey2019`, etc.)